crates/router/src/services/authentication.rs

```diff
@@ -90,6 +90,12 @@ pub struct AuthenticationDataWithUser {
     pub profile_id: Option<id_type::ProfileId>,
 }
 
+#[derive(Clone)]
+pub struct UserFromTokenWithRoleInfo {
+    pub user: UserFromToken,
+    pub role_info: authorization::roles::RoleInfo,
+}
+
```

Comment:
Can't we use `AuthenticationDataWithUser` which you've created earlier?

Reply:
No, we need `RoleInfo` here